src/SAML2/XML/saml/AuthenticatingAuthority.php

```diff
@@ -69,6 +66,9 @@ public function setAuthority(string $authority): void
      */
     public static function fromXML(DOMElement $xml): object
     {
+        Assert::same($xml->tagName, 'saml:AuthenticatingAuthority');
+        Assert::same($xml->namespaceURI, Constants::NS_SAML);
```

Comment:
Great! These checks should also be applied to the other elements, I think.

Reply:
I will do that

Reply:
I've added a commit for AuthnContext.. Can't do AuthnContextDecl because that one can contains virtually anything